### ISSUE TITLE
Improve error reporting

### ISSUE THREAD

**Issue by slippycheeze** [mbrt/gmailctl#53]
I've got a problem in my rules:
```
cannot parse config file: error parsing criteria for rule #20: empty filter node
```

...which is probably accurate, but having to carefully count forward 20 rules by hand, especially in the face of functions, is much more challenging than I'd like.

It'd be great if this dumped out the JSON or equivalent of the rule.  That won't give the original jsonnet for it, of course, but I'd expect it has a very high chance of including enough identifying details that would make it fast to track down the error.

This is an "application level" error, incidentally: the JSON generates correctly, but I'm presumably missing something in there.

**Comment by mbrt** [comment]
Makes sense. Errors are still quite cryptic, and I was going to improve that. What you are suggesting makes sense. The only catch is that I won't be able to give line numbers, unfortunately, because that context is lost in the Jsonnet -> JSON translation. But I guess some information is better than no information at all.

**Comment by mbrt** [comment]
Places to improve the reporting by including the contents of the broken filters:

* Filters imported from Gmail (see also comment in #49).
* Filters generated from the config (as initially reported).

**Comment by mbrt** [comment]
The exact problem reported here is addressed in #107. The more general issue to attach errors to line numbers, or make sure json and jsonnet syntax errors are reported in a saner way was not addressed because of limitations intrinsic in those packages.

I'll leave this open in case somebody has an idea on how to tackle those issues as well.
